addon/index.js

```diff
@@ -0,0 +1,31 @@
+import { decorator } from '@ember-decorators/utils/decorator-wrappers';
+import { run } from '@ember/runloop';
+
+function decoratorWithReturnValue(fn) {
+  return decorator(function(target, key, desc/*, params*/) {
+    // set as value on descriptor - a function that when called will 
+    // return run.next(myFunc)
+    return function(...args)  {
+      const { value } = desc;
+      return fn(value, ...args);
+    }
```

Comment:
oooh I forgot about the `context` here.  I wonder if the sol'n is just `() =>`?

Reply:
I think you could just pass `this` directly to the `fn` here:

```js
return fn(this, value, ...args)
```